### DISCUSSION TITLE
Solving PDE with time dependent parameter

### DISCUSSION THREAD

**Discussion by jiaor** [lululxvi/deepxde#1050] [Q&A]
Hi Lu et al.,

I am trying to solve the following equation.

$\frac {\partial A(r, t)}{\partial t} = \frac {D}{a^2} (\frac{\partial^2 A(r, t)}{\partial r^2} + \frac{2}{r} \frac{\partial A(r, t)}{\partial r}) + 1 $
in which $D = D_0 {\rm e}^ {-E_a/(RT)}$

It seems my code works when T is a constant but doesn't when T varies with t (model prediction differs from finite difference method). Can you help identify my problem and perhaps advice a solution?

```
D0 = 50e-4
a = 60e-6
Ea = 138e3
R = 8.3144621
D0a2 = D0 / a**2 * 365.25*24.*3600.*1e6
    
def pde(x, y):
    r = x[:, 0:1] 
    t = x[:, 1:2]
    # A = y[:, 0:1]

    T = (120. - 4. * t) + 273.15  #doesn't work
    # T = 60. + 273.15 #this works
    Da2 = D0a2 * tf.exp(-Ea/R/T)
    
    dA_r = dde.grad.jacobian(y, x, i=0, j=0)
    dA_t = dde.grad.jacobian(y, x, i=0, j=1)
    dA_rr = dde.grad.hessian(y, x, i=0, j=0)
    
    rhs0 = Da2 * (dA_rr + dA_r * 2/r) + 1
    func0 = dA_t - rhs0

    return func0

geom = dde.geometry.Interval(1e-3, 1)
timedomain = dde.geometry.TimeDomain(0, 30)
geomtime = dde.geometry.GeometryXTime(geom, timedomain)

def boundary_surface(x, on_boundary):
    return on_boundary and np.isclose(x[0], 1)

bc_surface = dde.icbc.DirichletBC(
    geomtime,
    lambda x: 0,
    boundary_surface
)

ic = dde.icbc.IC(
    geomtime,
    lambda x: 0,
    lambda _, on_initial: on_initial,
)

data = dde.data.TimePDE(
    geomtime, pde, [bc_surface, ic], num_domain=1000, num_boundary=80, num_initial=80,
)
net = dde.nn.FNN([2] + [20] * 3 + [1], "tanh", "Glorot normal")
model = dde.Model(data, net)

model.compile("adam", lr=1e-3)
model.train(iterations=1000)
model.compile("L-BFGS")
losshistory, train_state = model.train(iterations=1000)
# dde.saveplot(losshistory, train_state, issave=True, isplot=True)
```

**Comment by lululxvi** [comment]
What do you mean by "work" and "nor work"?

**Reply by jiaor** [reply]
Hi Lu,

I mean when T is constant the solution is the same as that I get from a finite difference method, but when T = f(t) I cannot get a solution close to the FDM. Sometime when T = f(t) has a very simple form the solution is a little bit better (closer to result from FDM) but requires many more iterations and sampling points. I am wondering if in addition to the time-dependent PDE, can I solve T = f(t) (or Da2 = f(D0a2) ) at the same time to make the process more efficient? Or is there another way I could transform the problem to make the learning process easier?

Thanks for your time!

**Reply by lululxvi** [reply]
I am not sure why you cannot get good solution for T = f(t). One possible issue is the scaling, and you can check FAQ.
